src/front/wgsl/lexer.rs

```diff
@@ -641,41 +385,195 @@ fn sub_test(source: &str, expected_tokens: &[Token]) {
     assert_eq!(lex.next().0, Token::End);
 }
 
+#[test]
+fn test_numbers() {
+    // WGSL spec examples //
+
+    // decimal integer
+    sub_test(
+        "0x123 0X123u 1u 123 0 0i 0x3f",
+        &[
+            Token::Number(Ok(Number::I32(291))),
```

Comment:
Unless these tests now need access to something private to `lexer.rs`, I'd rather they stayed in `tests.rs`. But I haven't tried this - if there's some reason it's difficult to have them in `tests.rs`, then that's fine. (Ideally, the tests in `lexer.rs` should move to `tests.rs`, too.)

Reply:
I mainly moved them here so that we can test the output and how the tokens are generated.

i.e.
`0x123` should be a concrete number literal with value `291` and not a number `0` followed by an `x` followed by a 123 number literal